xla/service/backend.h

```diff
@@ -125,13 +125,26 @@ class Backend {
   StatusOr<StreamPool::Ptr> BorrowStream(int device_ordinal);
   StatusOr<StreamPool::Ptr> BorrowStream(se::StreamExecutor* executor);
 
+  // Borrows a stream for use by the caller with a given priority.
```

Comment:
It seems 0 is the default priority in CUDA, and otherwise smaller number is higher priority (except 0)? I am thinking whether we should make the priority an enum instead of int to avoid any CUDA specific behavior, and also then not needing variants with priority as separate functions (i.e., set default value of priority to be the default enum). So enum Priority will be { 0=Default, Low, Med, High}?

Reply:
I added a new enum `StreamPriority` that has 3 levels, default, lowest and highest. I felt like Med might be a bit redundant since we have a default. So now all the cuda-specific stream creation is contained in gpu stream class.